client/lib/performance-tracking/components/performance-tracker-install.jsx

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable react/no-danger */
+
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+
+const performanceTrackerInstall = ( { nonce } ) => {
+	const serviceId = config( 'speedcurve_id' );
+	if ( ! config.isEnabled( 'rum-tracking/speedcurve' ) || ! serviceId ) return null;
+
+	return (
+		<>
+			<script
+				type="text/javascript"
+				nonce={ nonce }
+				dangerouslySetInnerHTML={ {
+					__html:
+						'LUX=(function(){var a=("undefined"!==typeof(LUX)&&"undefined"!==typeof(LUX.gaMarks)?LUX.gaMarks:[]);var d=("undefined"!==typeof(LUX)&&"undefined"!==typeof(LUX.gaMeasures)?LUX.gaMeasures:[]);var j="LUX_start";var k=window.performance;var l=("undefined"!==typeof(LUX)&&LUX.ns?LUX.ns:(Date.now?Date.now():+(new Date())));if(k&&k.timing&&k.timing.navigationStart){l=k.timing.navigationStart}function f(){if(k&&k.now){return k.now()}var o=Date.now?Date.now():+(new Date());return o-l}function b(n){if(k){if(k.mark){return k.mark(n)}else{if(k.webkitMark){return k.webkitMark(n)}}}a.push({name:n,entryType:"mark",startTime:f(),duration:0});return}function m(p,t,n){if("undefined"===typeof(t)&&h(j)){t=j}if(k){if(k.measure){if(t){if(n){return k.measure(p,t,n)}else{return k.measure(p,t)}}else{return k.measure(p)}}else{if(k.webkitMeasure){return k.webkitMeasure(p,t,n)}}}var r=0,o=f();if(t){var s=h(t);if(s){r=s.startTime}else{if(k&&k.timing&&k.timing[t]){r=k.timing[t]-k.timing.navigationStart}else{return}}}if(n){var q=h(n);if(q){o=q.startTime}else{if(k&&k.timing&&k.timing[n]){o=k.timing[n]-k.timing.navigationStart}else{return}}}d.push({name:p,entryType:"measure",startTime:r,duration:(o-r)});return}function h(n){return c(n,g())}function c(p,o){for(i=o.length-1;i>=0;i--){var n=o[i];if(p===n.name){return n}}return undefined}function g(){if(k){if(k.getEntriesByType){return k.getEntriesByType("mark")}else{if(k.webkitGetEntriesByType){return k.webkitGetEntriesByType("mark")}}}return a}return{mark:b,measure:m,gaMarks:a,gaMeasures:d}})();LUX.ns=(Date.now?Date.now():+(new Date()));LUX.ac=[];LUX.cmd=function(a){LUX.ac.push(a)};LUX.init=function(){LUX.cmd(["init"])};LUX.send=function(){LUX.cmd(["send"])};LUX.addData=function(a,b){LUX.cmd(["addData",a,b])};LUX_ae=[];window.addEventListener("error",function(a){LUX_ae.push(a)});LUX_al=[];if("function"===typeof(PerformanceObserver)&&"function"===typeof(PerformanceLongTaskTiming)){var LongTaskObserver=new PerformanceObserver(function(c){var b=c.getEntries();for(var a=0;a<b.length;a++){var d=b[a];LUX_al.push(d)}});try{LongTaskObserver.observe({type:["longtask"]})}catch(e){}};',
+				} }
+			/>
+			<script
+				src={ `https://cdn.speedcurve.com/js/lux.js?id=${ serviceId }` }
+				async
+				defer
+				crossOrigin="anonymous"
+			/>
+		</>
+	);
+};
+
+performanceTrackerInstall.propTypes = {
+	nonce: PropTypes.string,
```

Comment:
Should we add `isRequired`?

```suggestion
	nonce: PropTypes.string.isRequired,
```

Reply:
Turns out I can't make this required. This comes from https://github.com/Automattic/wp-calypso/blob/master/client/server/pages/index.js#L587, but that is only set for login pages (https://github.com/Automattic/wp-calypso/blob/master/client/server/pages/index.js#L572)

Reply:
Gotcha. Should we not be rendering when it's not provided?

Reply:
Sorry forgot to answer this.

From my understanding of the code it is ok to render if `nonce` is not provided and looks like we do render it in a few places:

![image](https://user-images.githubusercontent.com/975703/81040402-90e99380-8eab-11ea-95fb-e7f4fd5c19b9.png)